src/reason_parser.mly

```diff
@@ -3553,7 +3552,7 @@ constructor_declarations:
 ;
 
 constructor_declaration_no_leading_bar:
-  | as_loc(constr_ident) generalized_constructor_arguments attributes
+  | as_loc(UIDENT) generalized_constructor_arguments attributes
```

Comment:
I think you should also add a production here for `constr_ident` without `UIDENT`.